modules/importer/src/main/scala/ConfigReader.scala

```diff
@@ -38,72 +51,186 @@ object ConfigReader3 {
     val long:          Read[Long         ] = cast[Long]
     val offsetAngle:   Read[Angle        ] = string.map(s => Angle.fromArcsecs(s.toDouble))
-    val offsetP:       Read[OffsetP      ] = offsetAngle.map(OffsetP(_))
-    val offsetQ:       Read[OffsetQ      ] = offsetAngle.map(OffsetQ(_))
-    val instrument:    Read[Instrument   ] = enum(_.tccValue)
     val yesNo:         Read[Boolean      ] = cast[YesNoType].map(_.toBoolean)
     val durSecs:       Read[Duration     ] = double.map(_.toInt).map(Duration.ofSeconds(_))
-    val gcalLamp:      Read[GCalLamp     ] = cast[JSet[Object]].map(_.iterator.next.toString).map(ufindp[GCalLamp, String](_.tccValue))
-    val gcalShutter:   Read[GCalShutter  ] = seq[OldGCal.Shutter, GCalShutter](_.tccValue)
-    val f2fpu:         Read[F2FpUnit     ] = seq[OldF2.FPUnit,      F2FpUnit     ](_.tccValue)
-    val f2filter:      Read[F2Filter     ] = seq[OldF2.Filter,      F2Filter     ](_.tccValue)
-    val f2lyotwheel:   Read[F2LyotWheel  ] = seq[OldF2.LyotWheel,   F2LyotWheel  ](_.tccValue)
-    val f2disperser:   Read[F2Disperser  ] = seq[OldF2.Disperser,   F2Disperser  ](_.tccValue)
-
-    // It appears that the window cover is sometimes "bare" and someimes wrapped in `Some` ... it is
-    // not `None` in my test data so there's not a mapping for it.
-    val f2windowcover: Read[F2WindowCover] = { a =>
-      ufindp[F2WindowCover, String](_.tccValue) {
-        a match {
-          case a: OldF2.WindowCover        => a.sequenceValue
-          case s: GSome[OldF2.WindowCover] => s.getValue.sequenceValue
-        }
+  }
+
+
+  sealed abstract class System(private val system: String) {
+
+    override def toString: String =
+      s"System($system)"
+
+    final class Key[A](val name: String, val write: Show[A], val read: Read[A])(implicit ev: TypeTag[A]) {
+      val path = s"$system:$name"
+
+      def legacyGet(c: Map[String, AnyRef]): Option[AnyRef] =
+        c.get(path)
+
+      override def toString: String =
+        s"Key[${clean(ev.tpe.toString)}]($path)"
+
+      final class Entry private[Key](a: A) {
+        val value = a
+        val key   = Key.this
+        override def toString: String =
+          s"Key.Entry[${clean(ev.tpe.toString)}]($path -> ${write(a)})"
+      }
+
+      def apply(a: A) = new Entry(a)
+    }
+
+    protected object Key {
+      def apply[A: TypeTag](name: String, write: Show[A])(read: Read[A]): Key[A] =
+        new Key(name, write, read)
+
+      // Produces a pair of functions, one from old to new (S => A) and one
+      // from new to old (A => S)
+      private def xlat[S <: SequenceableSpType, A](name: String, m: Seq[(S, A)]): (S => A, A => S) = {
+        val oldToNew = m.toMap
+        val newToOld = m.map(_.swap).toMap
+
+        ((s: S) => oldToNew.getOrElse(s, sys.error(s"new value of '$name': $s not found")),
+         (a: A) => newToOld.getOrElse(a, sys.error(s"old value of '$name': $a not found")))
```

Comment:
Yeah, you're probably right.  We should keep it around for a little while regardless but at some point not too long after the transition we'll retire this code and stop maintaining it.